gstreamer is built with -Werror.  However on platforms that don't have
architecture specific implementations of the atomic functions, this
code would cause the compile to fail, so comment out the line.  This
is particularly important at the moment on IA-64.
					      -- agriffis, 13 Nov 2003

--- gstreamer-0.6.3/gst/gstatomic_impl.h.old	2003-08-03 16:52:19.000000000 -0400
+++ gstreamer-0.6.3/gst/gstatomic_impl.h	2003-11-13 14:43:34.000000000 -0500
@@ -415,7 +415,7 @@
 }
 
 #else 
-#warning consider putting your architecture specific atomic implementations here
+//#warning consider putting your architecture specific atomic implementations here
 
 /*
  * generic implementation
--- gstreamer-0.6.3/gst/elements/gstfilesrc.c.old	2003-11-14 13:41:30.000000000 -0500
+++ gstreamer-0.6.3/gst/elements/gstfilesrc.c	2003-11-14 14:28:12.000000000 -0500
@@ -396,8 +396,8 @@
     return NULL;
   }
   else if (mmapregion == MAP_FAILED) {
-    GST_DEBUG (0, "mmap (0x%x, %d, 0x%llx) : %s",
- 	     size, src->fd, offset, strerror (errno));
+    GST_DEBUG (0, "mmap (0x%lx, %ld, %d, 0x%llx) : %s",
+ 	     (unsigned long)NULL, size, src->fd, offset, strerror (errno));
     return NULL;
   }
   GST_DEBUG (0, "mapped region %08lx+%08lx from file into memory at %p", 
@@ -667,8 +667,7 @@
 
   /* check for EOF */
   if (src->curoffset == src->filelen) {
-    GST_DEBUG (0, "filesrc eos %" G_GINT64_FORMAT
-		  " %" G_GINT64_FORMAT, src->curoffset, src->filelen);
+    GST_DEBUG (0, "filesrc eos %llx %lld", src->curoffset, src->filelen);
     gst_element_set_eos (GST_ELEMENT (src));
     return GST_BUFFER (gst_event_new (GST_EVENT_EOS));
   }
@@ -836,19 +835,19 @@
           if (offset > src->filelen) 
 	    goto error;
           src->curoffset = offset;
-          GST_DEBUG(0, "seek set pending to %" G_GINT64_FORMAT, src->curoffset);
+          GST_DEBUG(0, "seek set pending to %llx", src->curoffset);
 	  break;
         case GST_SEEK_METHOD_CUR:
           if (offset + src->curoffset > src->filelen) 
 	    goto error;
           src->curoffset += offset;
-          GST_DEBUG(0, "seek cur pending to %" G_GINT64_FORMAT, src->curoffset);
+          GST_DEBUG(0, "seek cur pending to %llx", src->curoffset);
 	  break;
         case GST_SEEK_METHOD_END:
           if (ABS (offset) > src->filelen) 
 	    goto error;
           src->curoffset = src->filelen - ABS (offset);
-          GST_DEBUG(0, "seek end pending to %" G_GINT64_FORMAT, src->curoffset);
+          GST_DEBUG(0, "seek end pending to %llx", src->curoffset);
 	  break;
 	default:
           goto error;
